Non-static local-only function in async_context_XXX implementations

When running some other code with extra GCC warnings enabled (`-Wmissing-declaration`) it looks like one function in the `async_context_threadsafe_background` C-object should be `static`

````
In file included from /home/earle/Arduino/hardware/pico/rp2040/cores/rp2040/sdkoverride/async_context_threadsafe_background.c:2:
/home/earle/Arduino/hardware/pico/rp2040/cores/rp2040/sdkoverride/../../../pico-sdk/src/rp2_common/pico_async_context/async_context_threadsafe_background.c:138:10: warning: no previous declaration for 'async_context_threadsafe_background_execute_sync' [-Wmissing-declarations]
  138 | uint32_t async_context_threadsafe_background_execute_sync(async_context_t *self_base, uint32_t (*func)(void *param), void *param) {
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
````

This function is only used as a function pointer in the `async_context_type_t` struct and not defined in a header for external use (nor should it be, `async_context_execute_sync` is the proper interface).  Other functions assigned to this struct are all `static`, so it looks like just an oversight.

The same non-static state for only the execute function is present in `async_context_poll` and `async_context_freertos`.

--- a/src/rp2_common/pico_async_context/async_context_freertos.c
+++ b/src/rp2_common/pico_async_context/async_context_freertos.c
@@ -226,7 +226,7 @@ static void handle_sync_func_call(async_context_t *context, async_when_pending_w
     xSemaphoreGive(call->sem);
 }
 
-uint32_t async_context_freertos_execute_sync(async_context_t *self_base, uint32_t (*func)(void *param), void *param) {
+static uint32_t async_context_freertos_execute_sync(async_context_t *self_base, uint32_t (*func)(void *param), void *param) {
     async_context_freertos_t *self = (async_context_freertos_t*)self_base;
     hard_assert(xSemaphoreGetMutexHolder(self->lock_mutex) != xTaskGetCurrentTaskHandle());
     sync_func_call_t call = {0};

--- a/src/rp2_common/pico_async_context/async_context_poll.c
+++ b/src/rp2_common/pico_async_context/async_context_poll.c
@@ -51,7 +51,7 @@ static void async_context_poll_lock_check(async_context_t *self_base) {
     }
 }
 
-uint32_t async_context_poll_execute_sync(__unused async_context_t *context, uint32_t (*func)(void *param), void *param) {
+static uint32_t async_context_poll_execute_sync(__unused async_context_t *context, uint32_t (*func)(void *param), void *param) {
     return func(param);
 }
 

--- a/src/rp2_common/pico_async_context/async_context_threadsafe_background.c
+++ b/src/rp2_common/pico_async_context/async_context_threadsafe_background.c
@@ -135,7 +135,7 @@ static void lock_release(async_context_threadsafe_background_t *self) {
 #endif
 }
 
-uint32_t async_context_threadsafe_background_execute_sync(async_context_t *self_base, uint32_t (*func)(void *param), void *param) {
+static uint32_t async_context_threadsafe_background_execute_sync(async_context_t *self_base, uint32_t (*func)(void *param), void *param) {
     async_context_threadsafe_background_t *self = (async_context_threadsafe_background_t*)self_base;
 #if ASYNC_CONTEXT_THREADSAFE_BACKGROUND_MULTI_CORE
     if (self_base->core_num != get_core_num()) {

